ARCs/arc-0090.md

```diff
@@ -0,0 +1,527 @@
+description: Consolidated specification for encoding Algorand transactions and queries as URIs.
+author: Fabrice Benhamouda (@fabrice102), Ben Guidarelli (@barnjamin), Tasos Bitsios (@tasosbit), MG (@emg110), Barroso Stéphane (@sudoweezy)
+discussions-to: https://github.com/algorandfoundation/ARCs/discussions/359
+status: Draft
+type: Standards Track
+category: Interface
+sub-category: General
+created: 2022-04-21
+replaces: 26, 78, 79, 82
+---
+
+## Abstract
+
+This ARC defines a unified Algorand URI scheme that covers payment transactions, key registration, application NoOp calls, and read-only blockchain queries. It expands on earlier URI specifications to support deeplinks, QR codes, and other contexts where structured URIs communicate transaction intent or state queries.
+
+## Motivation
+
+This ARC consolidates and supersedes [ARC-26](./arc-0026.md), [ARC-78](./arc-0078.md), [ARC-79](./arc-0079.md), and [ARC-82](./arc-0082.md). Unifying their technical details avoids divergence across implementations, ensures extensions share consistent encoding rules, and provides a single reference for wallet, application, and tooling authors.
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in <a href="https://www.ietf.org/rfc/rfc2119.txt">RFC-2119</a>.
+
+Algorand URIs follow the general format for URIs as set forth in <a href="https://www.rfc-editor.org/rfc/rfc3986">RFC 3986</a>. The path component consists of an Algorand address, and the query component provides additional parameters specific to the encoded intent.
+
+### ABNF Overview
+
+The productions below consolidate the syntax for all Algorand URI variants. Scheme-specific sections reference these shared rules when describing their parameters.
+
+```abnf
+; Core
+algorandaddress   = *base32
+appid             = *digit
+assetid           = *digit
+; qchar corresponds to RFC 3986 query characters excluding "=" and "&"
+; qbase64url matches the unpadded base64url alphabet from RFC 4648 section 5
+qbase64url        = 1*(ALPHA / DIGIT / "-" / "_")
+alabel            = 1*(ALPHA / DIGIT / "-" / "_" / ".")
+noteparam         = "note=" *qchar
+xnote             = "xnote=" *qchar
+feeparam          = "fee=" *digit
+otherparam        = qchar *qchar [ "=" *qchar ]
+
+; Network authority selectors
+netauth           = ghlabel / netlabel
+ghlabel           = "gh:" 1*qbase64url
+netlabel          = "net:" ( "testnet" / "betanet" / alabel )
+
+; Payment transactions (ARC-26)
+paymenturn        = "algorand://" [ netauth "/" ] algorandaddress [ "?" paymentparams ]
+paymentparams     = paymentparam *( "&" paymentparam )
+paymentparam      = amountparam / labelparam / noteparam / xnote / assetparam / otherparam
+amountparam       = "amount=" *digit
+labelparam        = "label=" *qchar
+assetparam        = "asset=" *digit
+
```

Comment:
ARC26 is missing the sender definition.. can we add one of the params to be sender please?